streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -659,15 +632,13 @@ public boolean test(final K1 key, final V1 value) {
                                                    JoinWindows windows,
                                                    Serde<K1> keySerde,
                                                    Serde<V1> lhsValueSerde,
-                                                   Serde<V2> otherValueSerde,
-                                                   final String lhsStoreName,
-                                                   final String otherStoreName) {
+                                                   Serde<V2> otherValueSerde) {
 
             StateStoreSupplier thisWindow =
-                createWindowedStateStore(windows, keySerde, lhsValueSerde, lhsStoreName);
+                createWindowedStateStore(windows, keySerde, lhsValueSerde, name + "-this");
```

Comment:
Since the `name` here is the parent name, it may not reveal information for people looking at the store names, for example if you have `stream1.filter().join(stream2)` then the store names would be `KSTREAM-FILTER-xxx-this` and `KSTREAM-FILTER-xxx-other`, which are quite misleading right?

Instead, we could use for example `thisWindowStreamName + "-store"` and `otherWindowedStream + "-store"`.